### PR TITLE
Fix StoryArc & StoryArcNumber mismatch

### PR DESCRIPTION
**Pull request by fenrick**
# Fixed
- Fixed: Fixed processing logic error that if StoryArcNumber is empty, then multiple StoryArc's are not processed (similar would occur if multiple StoryArcNumber's are present for a single StoryArc).
- Fixed: Fixed inclusion of spaces in multiple StoryArc's due to ', ' as a common CSV format.

### CODE REVIEW

**Comment by fenrick** [comment]
These are two simple fixes that I was applying against the mainline code base, they were appropriate to contribute back. Edge cases for multiple StoryArc/StoryArcNumber are tricky to notice without Marvel/DC style comics with multiple StoryArcs overlapping in a single comic.

File: API/Services/ReadingListService.cs [hunk 1]
```diff
@@ -529,7 +537,7 @@ public async Task CreateReadingListsFromSeries(Series series, Library library)
                 arcNumbers[i] = int.MaxValue.ToString();
             }
             if (string.IsNullOrEmpty(arcs[i]) || !int.TryParse(arcNumbers[i], out _)) continue;
-            data.Add(new Tuple<string, string>(arcs[i], arcNumbers[i]));
+            data.Add(new Tuple<string, string>(arcs[i].Trim(), arcNumbers[i]));
```
Comment (majora2007) [inline API/Services/ReadingListService.cs:540]: Move arcs[i] into a variable and trim there, so you can use the !string.IsNullOrEmpty(arcs[i]) if statement appropriately.

File: API/Services/ReadingListService.cs [hunk 2]
```diff
@@ -519,6 +519,14 @@ public async Task CreateReadingListsFromSeries(Series series, Library library)
...
+        // ensure that both arrays are the same length
+        if (arcNumbers.Length < arcs.Length)
+        {
+            Array.Resize(ref arcNumbers, arcs.Length);
+        } else if (arcs.Length < arcNumbers.Length)
+        {
+            Array.Resize(ref arcs, arcNumbers.Length);
+        }
 
         var maxPairs = Math.Min(arcs.Length, arcNumbers.Length);
```
Comment (majora2007) [inline API/Services/ReadingListService.cs:531]: The goal here is to ensure we only work up until the max number of pairs. This line is now completely useless. In addition, I'm not sure what your resizing is actually doing, as from a logic perspective, we calculate the min number of pairs then process them. You're doing the same, but incuring a sys-call to allocate a new array and the memory along with it, which adds up when this is called per file.

**Review by majora2007 — CHANGES_REQUESTED** [review]
I'm not sure exactly what this PR is doing other than the Trim(). 

If you have files to reproduce, would you kindly share them with me via a DM (you can make them by removing all but 1 image from the file or sharing just the comicinfos).

**Comment by fenrick** [comment]
This is an example of a ComicInfo.xml generated by ComicTagger, based on ComicVine info - it contains three StoryArc's; but no matching StoryArcNumber's. When parsed by Kavita it only takes the first StoryArc.

```
<?xml version="1.0"?>
<ComicInfo xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <Title>Before Time Runs Out Part Two</Title>
  <Series>Avengers World</Series>
  <Number>18</Number>
  <Volume>2014</Volume>
  <StoryArc>Before Time Runs Out, Secret Wars, Time Runs Out</StoryArc>
  <Summary>The answers you’ve been waiting for—Before Time Runs Out!
Sunspot vs. A.I.M.—winner take all!
Starbrand and Nightmask face a surprising enemy!</Summary>
  <Notes>Tagged with ComicTagger 1.5.4 using info from Comic Vine on 2022-11-04 23:02:07.  [Issue ID 481606]</Notes>
  <Year>2015</Year>
  <Month>3</Month>
  <Day>4</Day>
  <Writer>Frank J. Barbiere</Writer>
  <Penciller>Marco Checchetto</Penciller>
  <Inker>Marco Checchetto</Inker>
  <Colorist>Andres Mossa</Colorist>
  <Letterer>Joe Caramagna</Letterer>
  <CoverArtist>Jeff Dekal</CoverArtist>
  <Editor>Jon Moisan, Tom Brevoort, Wil Moss</Editor>
  <Publisher>Marvel</Publisher>
  <Web>https://comicvine.gamespot.com/avengers-world-18-before-time-runs-out-part-two/4000-481606/</Web>
  <PageCount>25</PageCount>
  <Characters>Aleph, Andrew Forson, Black Swan, Cannonball, Captain America, Corvus Glaive, Emmanuel DaCosta, Invisible Woman, Maximus, Monica Rappaccini, Mr. Fantastic, Namor, Nightmask, Proxima Midnight, Star Brand, Sunspot, Superia, Terrax, Thanos</Characters>
  <Teams>A.I.M., Cabal, United Nations</Teams>
  <Locations>A.I.M. Island Base, Antigua and Barbuda, Barbuda, Rio de Janeiro, Space, The Necropolis, United Nations Headquarters, Wakanda</Locations>
  <Pages>
    <Page Image="0" ImageSize="372724" ImageWidth="1988" ImageHeight="3056" Type="FrontCover" />
  </Pages>
</ComicInfo>
```

File: API/Services/ReadingListService.cs (hunk 1 above)
Comment (fenrick) [reply API/Services/ReadingListService.cs:540]: Probably easier to switch is IsNullOrWhiteSpace as well.

File: API/Services/ReadingListService.cs (hunk 2 above)
Comment (fenrick) [reply API/Services/ReadingListService.cs:531]: I was attempted to honour the principle of the Math.Min statement, that was showing that there are circumstances where ArcNumbers might be larger then Arcs.

File: API/Services/ReadingListService.cs (hunk 1 above)
Comment (fenrick) [reply API/Services/ReadingListService.cs]: Reviewing the .net documentation has indicated a cleaner approach that I switched to. Using the Split to remove whitespaces and empty entries.

**Comment by majora2007** [comment]
@fenrick any chance you could implement the changes soon? I want to get this tested as I'm planning to release next release soon. If not, I can close this PR and make the changes for you.

**Review by majora2007 — APPROVED** [review]
Thanks for taking the effort to fix this and cleaning the code up.

**Pull request merged**